frontend/src/components/features/github/github-repo-selector.tsx

```diff
@@ -4,61 +4,80 @@ import { useDispatch } from "react-redux";
 import posthog from "posthog-js";
 import { setSelectedRepository } from "#/state/initial-query-slice";
 import { useConfig } from "#/hooks/query/use-config";
+import { useDebounce } from "#/hooks/use-debounce";
+import { useSearchRepositories } from "#/hooks/query/use-search-repositories";
 
 interface GitHubRepositorySelectorProps {
   onSelect: () => void;
   repositories: GitHubRepository[];
 }
 
+function sanitizeQuery(query: string) {
+  let sanitizedQuery = query.replace(/https?:\/\//, "");
+  sanitizedQuery = sanitizedQuery.replace(/github.com\//, "");
+  sanitizedQuery = sanitizedQuery.replace(/\.git$/, "");
+  sanitizedQuery = sanitizedQuery.toLowerCase();
+  return sanitizedQuery;
+}
+
 export function GitHubRepositorySelector({
   onSelect,
   repositories,
 }: GitHubRepositorySelectorProps) {
   const { data: config } = useConfig();
   const [selectedKey, setSelectedKey] = React.useState<string | null>(null);
+  const [searchQuery, setSearchQuery] = React.useState<string>("");
+  const debouncedSearchQuery = useDebounce(searchQuery, 300);
+  const { data: searchedRepos = [] } = useSearchRepositories(
```

Comment:
Instead of setting the default here, we can do so within the query itself through the `initialData` property

See https://tanstack.com/query/latest/docs/framework/react/guides/initial-query-data#using-initialdata-to-prepopulate-a-query